### PR TITLE
Additional user_settings.h examples

### PR DESCRIPTION
**Pull request by dgarske**

### CODE REVIEW

File: examples/configs/user_settings_template.h
Comment (lygstate) [inline examples/configs/user_settings_template.h:35]: #if !((defined(__MACH__) || defined(__FreeBSD__) || defined(__linux__) || defined(_WIN32)))

**Comment by lygstate** [comment]
template_all lack
```
   /* for static ciphers */
   #define WOLFSSL_STATIC_RSA
   #define WOLFSSL_STATIC_PSK
   #define WOLFSSL_STATIC_EPHEMERAL
   #define WOLFSSL_SNIFFER
```

**Comment by kaleb-himes** [comment]
Changes are good, just holding off on review due to merge conflicts that will likely arise with https://github.com/wolfSSL/wolfssl/pull/4124 refactor of the examples and addition of the testApps directory (Specifically the include.am file changes in this PR and in 4124)

**Comment by kaleb-himes** [comment]
Moving forward after discussing with @dgarske

**Review by kaleb-himes — APPROVED** [review]
Changes look good, thank you @dgarske, will merge once Jenkins finishes mulling them over.

File: examples/configs/user_settings_all.h
Comment (lygstate) [inline examples/configs/user_settings_all.h:216]: examples/configs/user_settings_template.h doesn't have these

**Pull request merged**
